framework/src/base/FEProblemBase.C

```diff
@@ -2081,6 +2142,13 @@ FEProblemBase::addInterfaceKernel(const std::string & interface_kernel_name,
     parameters.set<SubProblem *>("_subproblem") = this;
     parameters.set<SystemBase *>("_sys") = _nl.get();
   }
+
+  // Check that "variable" is in the NonlinearSystem.
+  if (!_nl->hasVariable(parameters.get<NonlinearVariableName>("variable")))
+    mooseError("Cannot add InterfaceKernel for variable ",
```

Comment:
In all `mooseErrors` you want to use the object name - it is coming in the `name` parameter - so that people know which block triggered the error. I suggest something like:
```
mooseError(name, ": error msg here");
```
Especially useful when you have tonnes of objects in your input file.